modules/search-service-impl/src/main/java/org/opencastproject/search/endpoint/SearchRestService.java

```diff
@@ -223,6 +223,47 @@ public Response remove(@PathParam("id") String mediaPackageId) throws SearchExce
     }
   }
 
+  @DELETE
+  @Path("removeSynchronously/{id}")
+  @Produces(MediaType.APPLICATION_XML)
+  @RestQuery(
+      name = "remove",
+      description = "Removes a mediapackage from the search index.",
+      pathParameters = {
+          @RestParameter(
+              description = "The media package ID to remove from the search index.",
+              isRequired = true,
+              name = "id",
+              type = RestParameter.Type.STRING
+          )
+      },
+      responses = {
+          @RestResponse(description = "OK", responseCode = HttpServletResponse.SC_OK),
+          @RestResponse(
+              description = "The mediapackage could not be delete by you or does not exist",
+              responseCode = HttpServletResponse.SC_BAD_REQUEST
+          ),
+          @RestResponse(
+              description = "There has been an internal error and the mediapackage could not be deleted",
+              responseCode = HttpServletResponse.SC_INTERNAL_SERVER_ERROR
+          )
+      },
+      returnDescription = "Nothing"
+  )
+  public Response removeSynchronously(@PathParam("id") String mediaPackageId) throws SearchException {
+    try {
+      boolean removed = searchService.deleteSynchronously(mediaPackageId);
```

Comment:
Oh boy.  You know what this *doesn't* do?  Update the index.  So you've removed it from search, but the admin ui doesn't know that :D

Reply:
Neither does `remove`, right? My understanding is that this is an internal API that would be used by some other part in the system to delete stuff from search, and then ALSO delete it from other places using APIs/services responsible for **those** places.

Reply:
Yeah, fair point now that you mention it.